lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1269,9 +1280,14 @@ def _private_key_generated(self) -> bool:
             return False
         return True
 
-    def _csr_matches_certificate_request(self, csr: CertificateSigningRequest) -> bool:
+    def _csr_matches_certificate_request(
```

Comment:
At the moment the is_ca field is a separate value in the charm relation interface databag. I agree that we should likely move to having it in the CSR, but this would impact the charm relation interface. There we would likely have to deprecate the field in the databag and encourage users to take the CSR route. 

This should go in a separate change.

Reply:
Alright.